tests/subsys/caf/sensor_data_aggregator/src/main.c

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <ztest.h>
+#include <app_event_manager.h>
+
+#include "test_events.h"
+#include <caf/events/sensor_event.h>
+#include "test_config.h"
+#include <zephyr/drivers/sensor.h>
+
+static enum test_id cur_test_id;
+static K_SEM_DEFINE(test_end_sem, 0, 1);
+
+
+void test_init(void)
+{
+	zassert_false(app_event_manager_init(), "Error when initializing");
+}
+
+static void test_start(enum test_id test_id)
+{
+	cur_test_id = test_id;
+	struct test_start_event *ts = new_test_start_event();
+
+	zassert_not_null(ts, "Failed to allocate event");
+	ts->test_id = test_id;
+	APP_EVENT_SUBMIT(ts);
+
+	int err = k_sem_take(&test_end_sem, K_SECONDS(30));
+
+	zassert_equal(err, 0, "Test execution hanged");
+}
+
+static void test_event_manager(void)
+{
+	test_start(TEST_EVENT_MANAGER);
+}
+
+static void test_basic(void)
+{
+	cur_test_id = TEST_BASIC;
+	struct test_start_event *ts = new_test_start_event();
+
+	zassert_not_null(ts, "Failed to allocate event");
+	ts->test_id = cur_test_id;
+	APP_EVENT_SUBMIT(ts);
+
+
+	size_t i = SAMPLES_IN_AGG_BUF * BASIC_TEST_AGG_EVENTS;
+
+	for (; i > 0; i--) {
+		struct sensor_event *se = new_sensor_event(sizeof(struct sensor_value) *
+			BASIC_TEST_SENSOR_SAMPLE_SIZE);
+
+		zassert_not_null(se, "Failed to allocate event");
+		se->descr = BASIC_TEST_AGG_DESCR;
+		se->dyndata.size = sizeof(struct sensor_value) * BASIC_TEST_SENSOR_SAMPLE_SIZE;
+		APP_EVENT_SUBMIT(se);
+		k_yield();
+	}
+
+	int err = k_sem_take(&test_end_sem, K_SECONDS(30));
+
+	zassert_equal(err, 0, "Test execution hanged");
+}
+
+static void test_order(void)
+{
+	cur_test_id = TEST_ORDER;
+	struct test_start_event *ts = new_test_start_event();
+
+	zassert_not_null(ts, "Failed to allocate event");
+	ts->test_id = cur_test_id;
+	APP_EVENT_SUBMIT(ts);
+
+	size_t i = SAMPLES_IN_AGG_BUF * ORDER_TEST_AGG_EVENTS;
+
+	for (; i > 0; i--) {
+		struct sensor_event *se = new_sensor_event(sizeof(struct sensor_value) *
+				ORDER_TEST_SENSOR_SAMPLE_SIZE);
+
+		zassert_not_null(se, "Failed to allocate event");
+		se->descr = ORDER_TEST_AGG_DESCR;
+		se->dyndata.size = sizeof(struct sensor_value) * ORDER_TEST_SENSOR_SAMPLE_SIZE;
+		se->dyndata.data[0] = i;
+		APP_EVENT_SUBMIT(se);
+		k_yield();
```

Comment:
This is something i wouldn't expect in the tests. If you are lacking of the HEAP just increase it. This looks like something is not good within the test.
Furthermore if the test is named `test_order` i would expect we are firstly flood the system with events and then we check if they came in correct order. With this yield you would probably create and receive event one by one. It kinda "tests" what event manager ensures - event order.